__tests__/spdx.test.ts

```diff
@@ -0,0 +1,267 @@
+import {expect, test} from '@jest/globals'
+import * as spdx from '../src/spdx'
+
+test('satisfiesAny', () => {
+  const units = [
+    {
+      candidate: 'MIT',
+      licenses: ['MIT'],
+      expected: true
+    },
+    {
+      candidate: 'MIT OR Apache-2.0',
+      licenses: ['MIT', 'Apache-2.0'],
+      expected: true
+    },
+    {
+      candidate: '(MIT AND ISC) OR Apache-2.0',
+      licenses: ['MIT', 'Apache-2.0'],
+      expected: true
+    },
+    {
+      candidate: 'MIT AND Apache-2.0',
+      licenses: ['MIT', 'Apache-2.0'],
+      expected: false
+    },
+    {
+      candidate: 'MIT AND BSD-3-Clause',
+      licenses: ['MIT', 'Apache-2.0'],
+      expected: false
+    },
+
+    // missing params, case sensitivity, syntax problems,
+    // or unknown licenses will return 'false'
+    {
+      candidate: 'MIT OR',
+      licenses: ['MIT', 'Apache-2.0'],
+      expected: false
+    },
+    {
+      candidate: '',
+      licenses: ['MIT', 'Apache-2.0'],
+      expected: false
+    },
+    {
+      candidate: 'MIT OR (Apache-2.0 AND ISC)',
+      licenses: [],
+      expected: false
+    },
+    {
+      candidate: 'MIT AND (ISC',
+      licenses: ['MIT', 'Apache-2.0'],
+      expected: false
+    },
+    {
+      candidate: 'MIT OR ISC',
+      licenses: ['MiT'],
+      expected: false
+    }
+  ]
+
+  for (const unit of units) {
+    let got: boolean = spdx.satisfiesAny(unit.candidate, unit.licenses)
+    if (got != unit.expected) {
+      console.log(
+        `failing unit test inputs: candidate(${unit.candidate}) licenses(${unit.licenses})`
+      )
+    }
```

Comment:
☝️ why log here? allowing expression parsing will make for some counter-intuitive matching results and I wanted the tests to illustrate this going forward, so I tried to set these up in a way that adding more trials would be easy as we address corner cases in the future.

To that end, I tried to install [jest-expect-message](https://www.npmjs.com/package/jest-expect-message) but it did not play nicely with Jest + TS, and after seeking some SME advice from the team, and noting in codesearch that I don't see prior art of this in use in similar Jest + TS projects, I gave up on that.

As-is, when these tests fail, you will see which case failed due to the conditional logging. If someone with more TS than I would like to 🍐 on using that instead, or just convert back to single test cases a la vanilla Jest, let me know 👍

Reply:
Agreed, I'm going to remove that.